test/minlp_tests.jl

```diff
@@ -0,0 +1,13 @@
+using MINLPTests
+
+const OPTIMIZER = MINLPTests.JuMP.with_optimizer(
+    Juniper.Optimizer, DefaultTestSolver()
+)
+
+@testset "MINLPTests" begin
+    MINLPTests.test_nlp_mi(OPTIMIZER, exclude = [
+        "005_011",  # Uses the function `\`
+        "003_013",  # Bug in Juniper - handling of expression graph?
+        "006_010"   # Bug in Juniper - handling of user-defined functions.
```

Comment:
Did you test `006_010`?  Maybe it works with the latest version of the solver?

Reply:
The problem there is that `Juniper.register` needs to be called as well. I'm not aware of a way to check whether JuMP.register was called inside Juniper and then add it to the internal data structure.

Reply:
Oh, interesting.  Lets merge this PR, but make an issue about asking Miles about how to avoid the need to call Juniper.register.

Reply:
He mentioned in #118:
> It looks like Juniper is trying to add an expression with the user-defined function to another JuMP model without registering it. However, there's no API to access the registry of user-defined functions from a JuMP model, so this is not an easy issue to fix.

Reply:
Post a feature request in JuMP.jl?  Now that we are on JuMP v0.19, it is worth our time to try to add features to that package.